src/coprocessor/dag/executor/mod.rs

```diff
@@ -219,9 +219,10 @@ pub fn build_exec(
 
 type FirstExecutor = (Box<Executor + Send>, Arc<Vec<ColumnInfo>>);
 
-fn build_first_executor(
+// TODO: Remove `'static` bound.
+fn build_first_executor<S: Snapshot + 'static>(
```

Comment:
@BusyJay I have to add `'static` bound here. Do you know why? The compiler does not give verbose enough information.

Reply:
Because the default lifetime of trait object is `'static`, so `TableScanExecutor/IndexScanExecutor` has to bound to 'static, which requires `S` bounds to `'static`.

Reply:
Wow! @busyjay thanks a lot!